components/email.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Manage BuddyPress Email Post Types.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Email extends BPCLI_Component {
+	/**
+	 * Create a new email post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --type=<type>
+	 * : Email type for the email (should be unique identifier, sanitized like a post slug).
+	 *
+	 * [--type-description=<type-description>]
+	 * : Email type description.
+	 *
+	 * --subject=<subject>
+	 * : Email subject line. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * --content=<content>
+	 * : Email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [--plain-text-content=<plain-text-content>]
+	 * : Plain-text email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp email create --type=new-event --type-description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --content="<a href='{{{some.custom-token-url}}}'></a>A new event</a> was created" --plain-text-content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		$term = term_exists( $assoc_args['type'], bp_get_email_tax_type() );
+
+		// Term already exists so don't do anything.
+		if ( $term !== 0 && $term !== null ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "Email type '{$assoc_args['type']}' already exists." );
+		}
+
+		if ( ! empty( $args[0] ) ) {
+			$assoc_args['content'] = $this->read_from_file_or_stdin( $args[0] );
+		}
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'edit' ) ) {
```

Comment:
There is no `edit` flag in this command doc. So either this is wrong or you need to add it to so that people can understand how it can be used.

Reply:
@r-a-y Could you say more about what this block does? I don't understand it from reading the code.

Reply:
I've fixed the `--edit` flag in commits 382cd27 and 2bd163b.

Basically, this works the same as the `wp post create` sub-command.  In order for this to work though, the `--content` option needs to be removed as a required option.

See how WP CLI does it here:
https://github.com/wp-cli/entity-command/blob/master/src/Post_Command.php#L37